src/nbody/preprocess.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+from typing import Any
+
+import yaml
+from body import Body
+
+# the following class is commented out because the module has not been developed yet
+# from generate_bodies import generate_random_bodies
+
+
+def read_simulation_config(filename: str) -> tuple[list[Body], dict[str, Any]]:
+    """Read bodies from yaml configuration file. If no bodies are provided,
+    generate random ones using an external generator."""
+
+    with open(filename) as f:
+        data = yaml.safe_load(f) or {}
```

Comment:
I think the code looks great! One suggestion I have is to add exception handling for the case when there is no .yaml file to be found. 
```suggestion
try:
    with open(filename) as f:
        data = yaml.safe_load(f) or {}
except FileNotFoundError:
    print(f"Error: The file '{filename}' was not found.")
```